tools/run_tests/python_utils/jobset.py

```diff
@@ -277,7 +277,10 @@ def start(self):
                 os.makedirs(logfile_dir)
             self._logfile = open(self._spec.logfilename, 'w+')
         else:
-            self._logfile = tempfile.TemporaryFile()
+            # macOS: a series of quick os.unlink invocation might cause OS
```

Comment:
did we see this failure before?  how does it exhibit itself? is there a link with more info about this issue?

Reply:
It occurs to me on my MacBook. I tried to twitch file handle limit, or remove temporary files. But it's unclear why the `os.unlink` failed with a system error.

Reply:
Right, I'm getting that too when I run `tools/buildgen/generate_projects.sh`
```
  File "tools/buildgen/generate_projects.py", line 98, in <module>
    err_cnt, _ = jobset.run(jobs, maxjobs=args.jobs)
  File "tools/buildgen/../run_tests/python_utils/jobset.py", line 589, in run
    if not js.start(cmdline):
  File "tools/buildgen/../run_tests/python_utils/jobset.py", line 474, in start
    self._quiet_success)
  File "tools/buildgen/../run_tests/python_utils/jobset.py", line 266, in __init__
    self.start()
  File "tools/buildgen/../run_tests/python_utils/jobset.py", line 280, in start
    self._logfile = tempfile.TemporaryFile()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/tempfile.py", line 624, in TemporaryFile
    _os.unlink(name)
```

But when I tried your fix, I'm now getting a warning:
```
Exception ignored in: <bound method _TemporaryFileCloser.__del__ of <tempfile._TemporaryFileCloser object at 0x7fc497733e10>>
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/tempfile.py", line 452, in __del__
    self.close()
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/tempfile.py", line 448, in close
    unlink(self.name)
InterruptedError: [Errno 4] Interrupted system call: '/var/folders/7w/1rl183bs1cz4ghlh8d6mnfhm006nph/T/tmp6_4e_41_'
```
The buildgen then passes, so there is an improvement, but it's not a 100% fix.